pkg/config/constants.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file describes the abstract model of services (and their instances) as
+// represented in Istio. This model is independent of the underlying platform
+// (Kubernetes, Mesos, etc.). Platform specific adapters found populate the
+// model object with various fields, from the metadata found in the platform.
+// The platform independent proxy code uses the representation in the model to
+// generate the configuration files for the Layer 7 proxy sidecar. The proxy
+// code is specific to individual proxy implementations
+
+package config
```

Comment:
Can we leave the top level pkg/config empty?
Ideally, I want to align this work with the Galley refactoring I am trying to do. IMO, ideally, we should have a single "pkg/config/..." that contains all config related things.

This can go to `pkg/config/constants` or `pkg/config/core`, for example?

Reply:
Sure that's reasonable, but I'll probably tackle that as a follow-on to this PR.  I'd like to further subdivide things into subpackages for labels, hostname, protocol, etc.  However, this will be a more invasive change throughout the codebase ... and this PR is already pretty complicated.

Reply:
sounds good.